server/api/oauthEnabledClient.js

```diff
@@ -66,55 +47,70 @@ const factory = ({ baseUrl, timeout }) => {
         })
         .timeout({ deadline: timeout / 3 })
         .end((error, response) => {
-          if (error) reject(superAgentErrorHandler(error))
-          else if (response) resolve(superAgentResultHandler(response))
+          if (error) reject(errorLogger(error))
+          else if (response) resolve(resultLogger(response))
         })
     })
 
   /**
-   * An Axios POST with Oauth token refresh and retry behaviour
+   * An superagent POST with Oauth token refresh and retry behaviour
    * @param context A request scoped context. Holds OAuth tokens and pagination information for the request
-   * @param url if url is relative then the baseURL will be prepended. If the url is absolute then baseURL is ignored.
+   * @param path relative path to post to.
    * @param body
-   * @returns A Promise which resolves to the Axios result object, or the Axios error object if it is rejected
+   * @returns A Promise which resolves to the superagent result object, or the superagent error object if it is rejected
    */
-  const post = (context, url, body) =>
-    axiosInstance(
-      addHeaders(context, {
-        method: 'post',
-        url,
-        data: body,
-      })
-    )
-      .then(resultLogger)
-      .catch(errorLogger)
-
-  const put = (context, url, body) =>
-    axiosInstance(
-      addHeaders(context, {
-        method: 'put',
-        url,
-        data: body,
-      })
-    )
-      .then(resultLogger)
-      .catch(errorLogger)
+  const post = (context, path, body) =>
+    new Promise((resolve, reject) => {
+      superagent
+        .post(baseUrl + path)
+        .send(body)
+        .set(getHeaders(context))
+        .end((error, response) => {
+          if (error) reject(errorLogger(error))
+          else if (response) resolve(resultLogger(response))
+        })
+    })
+  const put = (context, path, body) =>
+    new Promise((resolve, reject) => {
+      superagent
+        .put(baseUrl + path)
+        .send(body)
+        .set(getHeaders(context))
+        .end((error, response) => {
+          if (error) reject(errorLogger(error))
+          else if (response) resolve(resultLogger(response))
+        })
+    })
 
-  const getStream = (context, url) =>
-    axiosInstance(
-      addHeaders(context, {
-        method: 'get',
-        url,
-        responseType: 'stream',
-      })
-    ).catch(errorLogger)
+  const getStream = (context, path) =>
+    new Promise((resolve, reject) => {
+      superagent
+        .get(baseUrl + path)
+        .set(getHeaders(context))
+        .retry(2, (err, res) => {
+          if (err) logger.info(`Retry handler found API error with ${err.code} ${err.message}`)
+          return undefined // retry handler only for logging retries, not to influence retry logic
+        })
+        .timeout({ deadline: timeout / 3 })
+        .end((error, response) => {
+          if (error) reject(errorLogger(error))
+          else if (response) {
+            resultLogger(response)
```

Comment:
I couldn't work out a nice way of doing this.  Superagent suggests to call `pipe` instead of `end` to pass the data through.  This works fine when the image is found, but doesn't fail on images that aren't found.  Hence I read it into a node stream first and then response with the stream